### DISCUSSION TITLE
书面翻译内如何添加附录

### DISCUSSION THREAD

**Discussion by guoh064** [tuna/thuthesis#731] [Q&A] [answered]
如果在translation环境内直接使用`\appendix`命令的话标号会呈现".1"的状态而不是预想的“A.A"，不知道应该怎么实现，或者是我使用不当。

**Comment by xkwxdyy** [comment]
请提供一个还原你问题的示例文件

**Reply by guoh064** [reply]
[minimum.zip](https://github.com/tuna/thuthesis/files/8146478/minimum.zip)
我使用vscode的recipe: latexmk for thuthesis编译后效果如thuthesis-example.pdf所示，书面翻译的附录部分标号显示.1。
<img width="707" alt="image" src="https://user-images.githubusercontent.com/50830808/155841834-a6d4f378-90ae-439c-9c98-302755ed441f.png">

**Comment by zepinglee** [comment]
我感觉“A.A”表示“附录的附录”实在有些诡异。@Harry-Chen 怎么看？

**Comment by zepinglee** [comment]
我感觉“A.A”表示“附录的附录”实在有些诡异。@Harry-Chen 怎么看？

**Reply by Harry-Chen** [reply]
是有一些奇怪，但似乎没有更好的办法了？

**Reply by zepinglee** [reply]
那我实现一下吧。

**Comment by zepinglee** [comment, marked as answer]
@guoh064 试一下 https://github.com/tuna/thuthesis/tree/appendix 分支的效果有没有问题

**Reply by guoh064** [reply]
实现了上述需求，感谢。
（虽然这个标号确实诡异，如果有更好的表示方法倒也不必拘于此。）

我原来设想的是既然`abstract`环境在`translation`环境里的行为是“摘要”而非“中文摘要”、`abstract*`环境在`translation`环境里的行为同全篇论文的英文摘要，那么`abstract`应当是特别处理过的，可以在附录A中使用。
这时就得到了
[minimum2.zip](https://github.com/tuna/thuthesis/files/8147840/minimum2.zip)
中的特性：
<img width="706" alt="image" src="https://user-images.githubusercontent.com/50830808/155865136-5bf04b6e-8b55-48f5-b556-4d9b06000105.png">

现在采用appendix分支下的效果是这样的：
[minimum-appendix.zip](https://github.com/tuna/thuthesis/files/8147834/minimum-appendix.zip)
<img width="705" alt="image" src="https://user-images.githubusercontent.com/50830808/155865280-2c82b0d9-d726-4d15-8d14-9213fcd73701.png">

理想状态当然是“摘要、A.1、**附录A**、参考文献”的目录显示，但这究竟如何与文献翻译本身的标号相匹配我确实没有想明白，所以提这个问题的时候想的办法是A.A。开发者如果哪天想到了更好的方法也不必使用我这个。
最终还得看实践教学科有没有类似问题的解决方案。最好就是实践教学科在写作指南里直接提出“不写摘要附录”，这样大家都省心。

**Reply by guoh064** [reply]
我咨询了一下实践教学科，他们提到没有格式要求的部分就没有额外的格式要求了，并且附录里的摘要和附录（作为翻译的内容）也算字数，只要翻译的原文超过20000字就行。

**Comment by guoh064** [comment]
[minimum-appendix (2).zip](https://github.com/tuna/thuthesis/files/8159487/minimum-appendix.2.zip)

@zepinglee 非常抱歉，今天使用时发现了一个新的bug：目录的超链接以及pdf的书签超链接出现了问题，点击附录的A.A的超链接会跳转到A.1节，怀疑是实现appendix内标号的时候重置了计数器导致超链接生成有了问题，希望能修补一下，谢谢。
（为什么之前没发现呢，因为之前给的minimum是以附录A作为最后一页的，没有下一页也就没滚动了）

**Reply by zepinglee** [reply]
不必客气，试一下 8230bdf 的链接有没有别的问题？

**Reply by zepinglee** [reply]
另外建议贴出这样的示例就可以了，不必上传 zip。
```latex
\documentclass[degree=bachelor]{thuthesis}

\usepackage{lipsum}
\usepackage{hyperref}

\begin{document}

\appendix

\begin{translation}

\tableofcontents

\section{Section}
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]

\section{Section}
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]

\section{Section}
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]

\appendix
\section{Appendix}
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]

\section{Appendix}
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]
\subsection{Subsection}
\lipsum[1]

\end{translation}

\end{document}
```

**Reply by guoh064** [reply]
没有试出问题。感谢修补！
（下次一定仔细研究minimum代码的艺术）

**Reply by zepinglee** [reply]
那我就 merge 了。
